Updated Game Download Extension for Chrome

EnderInvader opened the pull request.

xtagon commented:
Hi @EnderInvader, since this PR replaces Cory's source to the same tool, would you mind summarizing for me what your version changes or adds to it?

EnderInvader commented:
> Hi @EnderInvader, since this PR replaces Cory's source to the same tool, would you mind summarizing for me what your version changes or adds to it?

Of course, it's essentially an overhauled version of Cory's tool:
- Updated the current game's JSON format
- Updated to the new chrome extension manifest V3 (V2 is now depreciated)
- Updated to new official api's which should provide better compatibility and security
- Updated backend to work a bit more fluidly
- New feature to copy a turn instead of just downloading it
- Completly new formatting and style


Old
![image](https://user-images.githubusercontent.com/20405891/175446267-c2c6b4b6-ef34-43fd-ab7f-3b05357610a7.png)

New
![image](https://user-images.githubusercontent.com/20405891/175445785-bdac1e82-045c-4635-b9a5-16874578685b.png)

xtagon commented:
Awesome! Thanks for contributing
